src/harness/fourslashInterfaceImpl.ts

```diff
@@ -1568,11 +1568,14 @@ namespace FourSlashInterface {
         ];
 
         export function globalsPlus(plus: readonly ExpectedCompletionEntry[]): readonly ExpectedCompletionEntry[] {
+            const firstEntry = plus[0];
+            const afterUndefined = typeof firstEntry !== "string" && firstEntry.sortText! > undefinedVarEntry.sortText!;

```

Comment:
Do you need a `!` for `undefinedVarEntry.sortText!`?

Reply:
Yeah, because its contextual (annotated) type has it optional. I’m actually knowingly possibly comparing `undefined` from `firstEntry.sortText` anyway, relying on the fact that any comparison involving `undefined` will evaluate to false.

Reply:
FWIW, the linter will error on unnecessary postfix-`!`